zebrad/tests/config.rs

```diff
@@ -0,0 +1,342 @@
+//! Integration tests for config loading via config-rs.
+//!
+//! Verifies layered configuration (defaults, TOML file, env) and
+//! `ZEBRA_`-prefixed environment variable mappings used in Docker or
+//! other environments.
+
+use std::{env, fs, io::Write, path::PathBuf, sync::Mutex};
+
+use tempfile::{Builder, TempDir};
+use zebrad::config::ZebradConfig;
+
+// Global mutex to ensure tests run sequentially to avoid env var races
+static TEST_MUTEX: Mutex<()> = Mutex::new(());
+
+/// Helper to isolate and manage ZEBRA_* environment variables in tests.
+struct EnvGuard {
+    _guard: std::sync::MutexGuard<'static, ()>,
+    original_vars: Vec<(String, String)>,
+}
+
+impl EnvGuard {
+    /// Acquire the global lock and clear all ZEBRA_* env vars, saving originals.
+    fn new() -> Self {
+        // If a test panics, the mutex guard is dropped, but the mutex remains poisoned.
+        // We can recover from the poison error and get the lock, because we're going
+        // to overwrite the environment variables anyway.
+        let guard = TEST_MUTEX.lock().unwrap_or_else(|e| e.into_inner());
+
+        let original_vars: Vec<(String, String)> = env::vars()
+            .filter(|(key, _val)| key.starts_with("ZEBRA_"))
```

Comment:
Could we move this prefix to a constant and use the constant here and in the `Drop` impl below?